sonar-scanner-engine/src/main/java/org/sonar/batch/report/ActiveRulesPublisher.java

```diff
@@ -38,16 +38,16 @@ public ActiveRulesPublisher(ActiveRules activeRules) {
   }
 
   @Override
-  public void publish(BatchReportWriter writer) {
-    Iterable<BatchReport.ActiveRule> activeRuleMessages = FluentIterable.from(activeRules.findAll()).transform(new ToMessage());
+  public void publish(ScannerReportWriter writer) {
+    Iterable<ScannerReport.ActiveRule> activeRuleMessages = FluentIterable.from(activeRules.findAll()).transform(new ToMessage());
     writer.writeActiveRules(activeRuleMessages);
   }
 
-  private static class ToMessage implements Function<ActiveRule, BatchReport.ActiveRule> {
-    private final BatchReport.ActiveRule.Builder builder = BatchReport.ActiveRule.newBuilder();
+  private static class ToMessage implements Function<ActiveRule, ScannerReport.ActiveRule> {
+    private final ScannerReport.ActiveRule.Builder builder = ScannerReport.ActiveRule.newBuilder();
 
     @Override
-    public BatchReport.ActiveRule apply(@Nonnull ActiveRule input) {
+    public ScannerReport.ActiveRule apply(@Nonnull ActiveRule input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)